File: examples/v2/agent-pools/go/main.go
package main

import (
	agents "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/agents"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		poolSuffix := cfg.Get("poolSuffix")
		if poolSuffix == "" {
			poolSuffix = "dev"
		}
		poolDescription := cfg.Get("poolDescription")
		if poolDescription == "" {
			poolDescription = "v2 example agent pool"
		}

		pool, err := agents.NewPool(ctx, "pool", &agents.PoolArgs{
			OrgName:     pulumi.String(serviceOrg),
			Name:        pulumi.String("v2-agent-pool-" + poolSuffix),
			Description: pulumi.String(poolDescription),
		})
		if err != nil {
			return err
		}

		ctx.Export("poolName", pool.Name)
		return nil
	})
}

File: examples/v2/agent-pools/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
pool_suffix = config.get("poolSuffix") or "dev"
pool_description = config.get("poolDescription") or "v2 example agent pool"

pool = ps_v2.agents.Pool(
    "pool",
    org_name=service_org,
    name=f"v2-agent-pool-{pool_suffix}",
    description=pool_description,
)

pulumi.export("poolName", pool.name)
